tools/ci_build/build.py

```diff
@@ -675,6 +683,12 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
                        "-Deigen_SOURCE_PATH=" + args.eigen_path]
 
     if args.android:
+        if args.use_dnnlibrary and args.use_nnapi:
+            raise BuildError(
+                "Only one of --use_dnnlibrary and --use_nnapi " +
+                "can be enabled"
+            )
+
```

Comment:
Should we also check that both aren't enabled in CMakeLists.txt? It's possible to build without using this script.